src/functionalTest/java/uk/gov/hmcts/reform/cwrdapi/CaseWorkerRefFunctionalTest.java

```diff
@@ -437,17 +439,19 @@ public void deleteCaseworkerById() {
         assertThat(fetchResponse.getStatusCode()).isEqualTo(404);
     }
 
-    @Test
+    @ParameterizedTest
+    @ValueSource(strings = {""})
     @ToggleEnable(mapKey = DELETE_CASEWORKER_BY_ID_OR_EMAILPATTERN, withFeature = true)
     @ExtendWith(FeatureToggleConditionExtension.class)
     //this test verifies that a User Profile is deleted by Email Pattern
-    public void deleteCaseworkerByEmailPattern() {
-        String emailPattern = "deleteTest1234";
-        String email = format(EMAIL_TEMPLATE, randomAlphanumeric(10) + emailPattern).toLowerCase();
+    public static void deleteCaseworkerByEmailPattern(String emailPattern) {
 
         List<CaseWorkersProfileCreationRequest> caseWorkersProfileCreationRequests =
-                caseWorkerApiClient.createCaseWorkerProfiles(email);
+                createNewActiveCaseWorkerProfile();
 
+        if (emailPattern.isEmpty()) {
+            emailPattern = caseWorkersProfileCreationRequests.get(0).getEmailId();
+        }
```

Comment:
extract delete endpoint logic to separate method and use that method in this test case and also after all cleanUpTestData method, and you don't need ParameterizedTest and ValueSource